src/calico_manifests.py

```diff
@@ -375,3 +381,62 @@ def config_hash(self) -> str:
         hash = hashlib.sha256()
         hash.update(json_str.encode())
         return hash.hexdigest()
+
+    def status(self) -> FrozenSet[HashableResource]:
+        """Return all installed objects which have a `.status.conditions` attribute.
+
+        Bonus: Log events for daemonsets with unready pods.
+        """
+        installed = self.installed_resources()
+        for obj in installed:
+            if obj.kind == "DaemonSet":
+                ds: DaemonSet = obj.resource
+                if ds.status.numberReady != ds.status.desiredNumberScheduled:
+                    log_events(collect_events(self.client, obj.resource))
+        return frozenset(_ for _ in installed if _.status_conditions)
```

Comment:
overrides the base `Manifest.status(...)` to return the same results, but with a bonus of logging daemonset events if its in failures. 

We can't do this on `Manifest.is_ready(...)` because the calico daemonsets never create a `status.conditions` list